Assets/Scripts/UI/Cutscenes/CutsceneController.cs

```diff
@@ -0,0 +1,151 @@
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using System.Collections.Generic;
+using OperationBlackwell.Core;
+
+namespace OperationBlackwell.UI {
+	public class CutsceneController : BaseCutsceneController {
+		public static CutsceneController Instance { get; private set; }
+
+		[SerializeField] private Cutscene[] cutscenes_;
+		private Cutscene currentCutscene_;
+		private int currentCutsceneIndex_;
+		private int currentMessageIndex_;
+
+		private List<GameObject> children_;
+		private GameObject base_;
+		private GameObject leftButton_;
+		private GameObject rightButton_;
+		private GameObject endButton_;
+		private Image leftChar_;
+		private Image rightChar_;
+		private TextMeshProUGUI leftName_;
+		private TextMeshProUGUI rightName_;
+		private TextMeshProUGUI speakingName_;
+		private TextMeshProUGUI text_;
+
+		private void Awake() {
+			if (Instance == null) {
+				Instance = this;
+			} else {
+				Destroy(gameObject);
+				return;
+			}
+			children_ = new List<GameObject>();
+		}
+
+		private void Start() {
+			currentCutscene_ = null;
+			currentCutsceneIndex_ = 0;
+			foreach(Transform child in transform) {
+				children_.Add(child.gameObject);
+			}
+			base_ = children_[0];
+			leftButton_ = children_[1];
+			rightButton_ = children_[2];
+			endButton_ = children_[3];
+			leftChar_ = children_[4].GetComponent<Image>();
+			rightChar_ = children_[5].GetComponent<Image>();
+			leftName_ = children_[6].GetComponent<TextMeshProUGUI>();
+			rightName_ = children_[7].GetComponent<TextMeshProUGUI>();
+			speakingName_ = children_[8].GetComponent<TextMeshProUGUI>();
+			text_ = children_[9].GetComponent<TextMeshProUGUI>();
+
+			SetActive(false);
+
+			// This is here for testing now, need to remove this later.
+			StartCutscene();
+		}
+
+		private void Show(int index) {
+			currentCutscene_ = cutscenes_[index];
+			currentMessageIndex_ = 0;
+			SelectNextText();
+			SetActive(true);
+			SetChildActive(leftButton_, false);
+			SetChildActive(endButton_, false);
+		}
+
+		private void Hide() {
+			currentCutscene_ = null;
+			SetActive(false);
+		}
+
+		public override void StartCutscene() {
+			GridCombatSystem.Instance.SetState(GridCombatSystem.State.Cutscene);
+			Show(currentCutsceneIndex_);
+		}
+
+		public override void EndCutscene() {
+			Hide();
+			GridCombatSystem.Instance.SetState(GridCombatSystem.State.Normal);
+			currentCutsceneIndex_++;
+		}
+
+		private void SetActive(bool active) {
+			foreach(GameObject child in children_) {
+				child.SetActive(active);
+			}
+		}
+
+		private void SetChildActive(GameObject child, bool active) {
+			child.SetActive(active);
+		}
+
+		private void SelectNextText() {
```

Comment:
This can only handle one character on each side right? I'm unsure if that feature is needed but something to keep in mind.

Reply:
This was already decided by design to have one on each side.

Reply:
Then this is fine.